### PR TITLE
Use Testcontainers instead of Docker Compose for tests with PostgreSQL

### PR DESCRIPTION
**Pull request by odzhychko**
Use Testcontainers instead of Docker Compose for tests with PostgreSQL.
This also made the separate `model-server-test` project obsolete.

## To be verified by reviewers

* [ ] Relevant public API members have been documented
* [ ] Documentation related to this PR is complete
  * [ ] Boundary conditions are documented
  * [ ] Exceptions are documented
  * [ ] Nullability is documented if used
* [ ] Touched existing code has been extended with documentation if missing
* [ ] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

File: model-server/src/test/kotlin/org/modelix/model/server/store/IgnitePostgresRepositoryRemovalTest.kt [hunk 1]
```diff
@@ -13,33 +13,52 @@
...
     private val toDelete = RepositoryId("repository-removal-toDelete")
     private val existing = RepositoryId("repository-removal-existing")
 
+    @BeforeTest
+    fun beforeTest() {
+        postgres = PostgreSQLContainer(DockerImageName.parse("postgres:16.2-alpine"))
+            .withDatabaseName("modelix")
+            .withUsername("modelix")
+            .withPassword("modelix")
+            .withExposedPorts(5432)
```
Comment (slisson) [inline model-server/src/test/kotlin/org/modelix/model/server/store/IgnitePostgresRepositoryRemovalTest.kt:47]: Chances are high that this port is already used and the test will fail. Tests should use some random non-default port.

File: model-server/src/main/kotlin/org/modelix/model/server/Main.kt [hunk 2]
```diff
@@ -251,6 +255,16 @@ object Main {
         }
     }
 
+    private fun readJdbcProperties(jdbcConfFile: File): Properties {
+        val properties = Properties()
+        try {
+            properties.load(FileReader(jdbcConfFile))
+        } catch (e: IOException) {
+            throw IllegalStateException("Could to load the JDBC configuration from ${jdbcConfFile.absolutePath}.", e)
```
Comment (slisson) [inline model-server/src/main/kotlin/org/modelix/model/server/Main.kt:263]:
```suggestion
            throw IllegalStateException("Could not load the JDBC configuration from ${jdbcConfFile.absolutePath}.", e)
```

File: model-server/src/test/kotlin/org/modelix/model/server/store/IgnitePostgresRepositoryRemovalTest.kt (hunk 1 above)
Comment (odzhychko) [reply model-server/src/test/kotlin/org/modelix/model/server/store/IgnitePostgresRepositoryRemovalTest.kt:47]: `withExposedPorts` exposes the specified port from the container to a random free port on the host.

The port on the host is retrieved by calling `postgres.firstMappedPort` in line 51.

Comment (slisson) [reply model-server/src/test/kotlin/org/modelix/model/server/store/IgnitePostgresRepositoryRemovalTest.kt:47]: Nice.

**Review by slisson — APPROVED** [review]
